packages/deployment/test/test_lambda.js

```diff
@@ -87,7 +87,42 @@ test.serial('zipLambda: works for lambda using message adapter', async (t) => {
 });
 
 test.serial(
-  `zipLambda: for lambda using message adapter, no new file is generated 
+  `zipLambda: given an invalid zip file generated from a previous run,
+  a new valid lambda file is generated`,
+  async (t) => {
+    t.context.lambda.useMessageAdapter = true;
+    const lambdaLocalOrigin = t.context.lambda.local;
+    const lambdaRemoteOrigin = t.context.lambda.remote;
+
+    // put an empty lambda zip file there as the result of the previous run
+    const existingLambdaLocal = path.join(
+      path.dirname(t.context.lambda.local),
+      `${Lambda.messageAdapterZipFileHash}-${path.basename(t.context.lambda.local)}`
+    );
+
+    fs.writeFileSync(existingLambdaLocal, 'hello');
+    t.is(fs.statSync(existingLambdaLocal).size, 5);
```

Comment:
Same comment as above, consider using the Promise-returning version from fs-extra.